homeassistant/components/connectedcars/config_flow.py

```diff
@@ -0,0 +1,136 @@
+"""Config flow for ConnectedCars.io integration."""
+import logging
+
+from connectedcars import ConnectedCarsClient, ConnectedCarsException
+import nest_asyncio
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import (  # pylint:disable=unused-import
+    COMPLETE_QUERY_USER,
+    COMPLETE_QUERY_VIN,
+    DEFAULT_NAMESPACE,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("namespace", default=DEFAULT_NAMESPACE): str,
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+        vol.Optional("vin"): str,
+    }
+)
+
+
+class ConnectedcarsApiHandler:
+    """Connected Cars.io API connector class."""
+
+    def __init__(self, namespace):
+        """Initialize."""
+        self.namespace = namespace
+        self.vin = None
+        self.client = None
+        self.user_data = None
+
+    async def authenticate(self, username, password) -> bool:
+        """Test if we can authenticate with the namespace."""
+        client = ConnectedCarsClient(username, password, self.namespace)
+
+        self.client = client
+
+        self.user_data = await client.async_query(COMPLETE_QUERY_USER)
+
+        return True
+
+    async def get_email(self) -> str:
+        """Will get the email address of the account holder."""
+
+        if not self.user_data:
+            self.user_data = await self.client.async_query(COMPLETE_QUERY_USER)
+
+        email = self.user_data["data"]["viewer"]["email"]
+
+        return email
+
+    async def get_vin(self, userinput_vin) -> str:
+        """Will get the vin identifier of the car."""
+
+        if not self.vin:
+            response = await self.client.async_query(COMPLETE_QUERY_VIN)
+            # This needs to lookup the "userinput_vin" to see if it can be found in the result, not just take the first car
+            vin = response["data"]["viewer"]["vehicles"][0]["vehicle"]["vin"]
+            self.vin = vin
+
+        return self.vin
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    nest_asyncio.apply()
```

Comment:
I don't think we can allow patching asyncio. Why are we doing that here?

Reply:
Thanks @MartinHjelmare. This is because the connectedcars pypi package that i use are using asyncio, and that gave me some problems with the scoping of them.
nest_asyncio was a workaround i found on the internet and it seemed to work.
If this is not the right way, then I will go back and check for another solution. I am, however, not maintaining the pypi package that I am using, so I would possibly have to create my own version of that package then; making this a bigger task then expected.
So is it not allowed to use nest_asyncio?